### PR TITLE
Feat/normalize route

### PR DESCRIPTION
**Pull request by hilaYarkoni**

### CODE REVIEW

File: lib/utils/normalizers.ts [hunk 1]
```diff
@@ -1,16 +1,78 @@
...
   return urlParser.replacePathValues(pathname, placeholder);
 }
 
 export function normalizeStatusCode(statusCode: number): string {
-  if (statusCode >= 200 && statusCode < 300) return "2XX";
-  if (statusCode >= 300 && statusCode < 400) return "3XX";
-  if (statusCode >= 400 && statusCode < 500) return "4XX";
+  if (statusCode >= 200 && statusCode < 300) return '2XX';
+  if (statusCode >= 300 && statusCode < 400) return '3XX';
+  if (statusCode >= 400 && statusCode < 500) return '4XX';
```
Comment (odedino) [inline lib/utils/normalizers.ts:17]: I wouldn't normalize it - it has no real value as we can do it in the query to prometheus

File: lib/utils/normalizers.ts [hunk 2]
```diff
@@ -1,16 +1,78 @@
...
+    ) {
+      // normalizedRoutePath = originalUrl.split('?')[0];
+      // in case the requested route doesn't exist
+      return 'unknown';
+    } else {
+      const splittedRoute = normalizedRoutePath.split('/');
+      const splittedUrl = originalUrl.split('?')[0].split('/');
+      const routeIndex = splittedUrl.length - splittedRoute.length + 1;
+      const baseUrl = splittedUrl.slice(0, routeIndex).join('/');
+      normalizedRoutePath = baseUrl + normalizedRoutePath;
```
Comment (odedino) [inline lib/utils/normalizers.ts:59]: You removed the query params but as you can see in the original code - it was configurable - so I suggest to return it and have it off by default but allow to set it:

see `this.setupOptions.includeQueryParams` 

```
if (this.setupOptions.includeQueryParams === true && Object.keys(req.query).length > 0) {
                route = `${route}?${Object.keys(req.query).sort().map((queryParam) => `${queryParam}=<?>`).join('&')}`;
            }
```

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@marbletech/nestjs-prom",
   "version": "1.0.0",
-  "description": "A promotheus module for nestjs",
+  "description": "A Marble/rnd promotheus module for nestjs",
```
Comment (odedino) [inline package.json]: I wouldn't change it if we plan to merge it to the original code

File: lib/middleware/inbound.middleware.ts [hunk 4]
```diff
@@ -25,15 +27,18 @@ export class InboundMiddleware implements NestMiddleware {
     });
   }
 
-  use (req, res, next) {
+  use(req, res, next) {
     responseTime((req, res, time) => {
       const { url, method } = req;
-      const path = normalizePath(url, this._options.withHttpMiddleware?.pathNormalizationExtraMasks, "#val");
```
Comment (odedino) [inline lib/middleware/inbound.middleware.ts:31]: NIT - might be worth keeping the options of the current lib as  passing extra masks ( if it makes sense), and  ` `placeholder` - easy to acheieve

**Review by odedino — APPROVED** [review]

File: lib/utils/normalizers.ts (hunk 2 above)
Comment (hilaYarkoni) [reply lib/utils/normalizers.ts:59]: Can I pass it to the normalize method? 
`export function normalizeRoute(req, includeQueryParams = false)`

**Pull request merged**